torch/autograd/_functions/pointwise.py

```diff
@@ -331,93 +362,90 @@ class Remainder(_ConstantGrad):
         else:
-            return add_tensor.addcmul(self.scale, mul_tensor1, mul_tensor2)
+            return add_tensor.addcmul(scale, mul_tensor1, mul_tensor2)
 
-    def backward(self, grad_output):
+    @staticmethod
+    def backward(ctx, grad_output):
         grad_add = grad_mul1 = grad_mul2 = None
-        mul_tensor1, mul_tensor2 = self.saved_tensors
+        mul_tensor1, mul_tensor2 = ctx.saved_variables
 
-        if self.needs_input_grad[0]:
+        if ctx.needs_input_grad[0]:
             grad_add = grad_output
 
-        if self.needs_input_grad[1]:
-            grad_mul1 = grad_output.mul(mul_tensor2).mul(self.scale)
+        if ctx.needs_input_grad[1]:
+            grad_mul1 = grad_output.mul(mul_tensor2).mul(ctx._scale)
 
-        if self.needs_input_grad[2]:
-            grad_mul2 = grad_output.mul(mul_tensor1).mul(self.scale)
+        if ctx.needs_input_grad[2]:
+            grad_mul2 = grad_output.mul(mul_tensor1).mul(ctx._scale)
 
-        return grad_add, grad_mul1, grad_mul2
+        return grad_add, grad_mul1, grad_mul2, None, None
 
 
 class Addcdiv(InplaceFunction):
 
-    def __init__(self, scale=1, inplace=False):
-        super(Addcdiv, self).__init__(inplace)
-        self.scale = scale
-
-    def forward(self, add_tensor, div_tensor1, div_tensor2):
-        self.save_for_backward(div_tensor1, div_tensor2)
-        if self.inplace:
-            self.mark_dirty(add_tensor)
-            return add_tensor.addcdiv_(self.scale, div_tensor1, div_tensor2)
+    @staticmethod
+    def forward(ctx, add_tensor, div_tensor1, div_tensor2, scale=1.0, inplace=False):
+        ctx._scale = scale
+        ctx.save_for_backward(div_tensor1, div_tensor2)
+        if inplace:
+            ctx.mark_dirty(add_tensor)
+            return add_tensor.addcdiv_(ctx._scale, div_tensor1, div_tensor2)
         else:
-            return add_tensor.addcdiv(self.scale, div_tensor1, div_tensor2)
+            return add_tensor.addcdiv(ctx._scale, div_tensor1, div_tensor2)
 
-    def backward(self, grad_output):
+    @staticmethod
+    def backward(ctx, grad_output):
         grad_add = grad_div1 = grad_div2 = None
-        div_tensor1, div_tensor2 = self.saved_tensors
+        div_tensor1, div_tensor2 = ctx.saved_variables
 
-        if self.needs_input_grad[0]:
+        if ctx.needs_input_grad[0]:
             grad_add = grad_output
 
-        if self.needs_input_grad[1]:
-            grad_div1 = grad_output.div(div_tensor2).mul(self.scale)
+        if ctx.needs_input_grad[1]:
+            grad_div1 = grad_output.div(div_tensor2).mul(ctx._scale)
 
-        if self.needs_input_grad[2]:
+        if ctx.needs_input_grad[2]:
             div_tensor2_sq = div_tensor2.mul(div_tensor2)
-            grad_div2 = grad_output.mul(div_tensor1).div_(div_tensor2_sq)
-            grad_div2.neg_().mul_(self.scale)
-
-        return grad_add, grad_div1, grad_div2
+            grad_div2 = grad_output.mul(div_tensor1).div(div_tensor2_sq).neg().mul(ctx._scale)
```

Comment:
So, some in-place ops in Variables can be used without worrying too much? I suppose this only applies to invertible ops, like `neg_`?